distribution/xfer/download.go

```diff
@@ -33,11 +36,12 @@ func (ldm *LayerDownloadManager) SetConcurrency(concurrency int) {
 }
 
 // NewLayerDownloadManager returns a new LayerDownloadManager.
-func NewLayerDownloadManager(layerStore layer.Store, concurrencyLimit int, options ...func(*LayerDownloadManager)) *LayerDownloadManager {
+func NewLayerDownloadManager(layerStore layer.Store, concurrencyLimit int, cacheRoot string, options ...func(*LayerDownloadManager)) *LayerDownloadManager {
```

Comment:
Just define a new option type for `cacheRoot`. `concurrencyLimit` should have had that done, as well.